### PR TITLE
Consider Nightly channels when checking for avaliable_channels

### LINKED ISSUE #820: Error says: "Invalid Juliaup channel `nightly`"
Huzzah! `nightly` is now a valid channel. This is so exciting!

However, when the `nightly` channel is not installed, it is mischaracterized as "invalid" instead of "not installed":
```
x@fedora:~$ julia +1.9.0
`1.9.0` is not installed. Please run `juliaup add 1.9.0` to install channel or version.
x@fedora:~$ julia +nightly
ERROR: Invalid Juliaup channel `nightly`. Please run `juliaup list` to get a list of valid channels and versions.
x@fedora:~$ juliaup list
 Channel                Version                           
----------------------------------------------------------
 0.6.1                  0.6.1+0.aarch64.linux.gnu         
 0.6.1~aarch64          0.6.1+0.aarch64.linux.gnu         
 0.6.2                  0.6.2+0.aarch64.linux.gnu         
 0.6.2~aarch64          0.6.2+0.aarch64.linux.gnu         
 ...
 beta~aarch64           1.10.0+0.aarch64.linux.gnu        
 lts                    1.6.7+0.aarch64.linux.gnu         
 lts~aarch64            1.6.7+0.aarch64.linux.gnu         
 rc                     1.10.0+0.aarch64.linux.gnu        
 rc~aarch64             1.10.0+0.aarch64.linux.gnu        
 release                1.10.0+0.aarch64.linux.gnu        
 release~aarch64        1.10.0+0.aarch64.linux.gnu        
 nightly                latest-linux-aarch64              
 nightly~aarch64        latest-linux-aarch64              
x@fedora:~$ juliaup add nightly
Installing Julia latest-linux-aarch64
x@fedora:~$ julia +nightly --startup-file=no
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.11.0-DEV.1458 (2024-02-02)
 _/ |\__'_|_|_|\__'_|  |  Commit 98e4f01d26e (0 days old master)
|__/                   |

julia> 
```

### PR DESCRIPTION
**Pull request by christiangnrd**
Closes #820 but not #995.

### CODE REVIEW

**Comment by christiangnrd** [comment]
This is ready for review.

File: src/operations.rs [hunk 1]
```diff
@@ -431,6 +431,31 @@ pub fn compatible_archs() -> Result<Vec<String>> {
     }
 }
 
+// which nightly channels are compatible with the current system
+fn compatible_nightly_channels() -> Result<Vec<String>> {
+    let archs: Result<Vec<String>> = compatible_archs();
```
Comment (davidanthoff) [inline src/operations.rs:436]: I think if you just terminate this line with a `?` then you can drop the whole `if` `else` construction later in the function.

File: src/operations.rs [hunk 2]
```diff
@@ -431,6 +431,31 @@ pub fn compatible_archs() -> Result<Vec<String>> {
...
+            .collect();
+        Ok(channels)
+    } else {
+        archs
+    }
+}
+
+// considers the nightly channels as system channels
+// XXX: does not account for PR channels
+pub fn is_valid_channel(versions_db: &JuliaupVersionDB, channel: &String) -> bool {
```
Comment (davidanthoff) [inline src/operations.rs:450]: Return type should also be a `Result`, right?

File: src/operations.rs [hunk 3]
```diff
@@ -431,6 +431,31 @@ pub fn compatible_archs() -> Result<Vec<String>> {
...
+    let regular = versions_db.available_channels.contains_key(channel);
+
+    let nightly_chans = compatible_nightly_channels();
```
Comment (davidanthoff) [inline src/operations.rs:453]: And then terminate with `?`

**Review by davidanthoff — CHANGES_REQUESTED** [review]
Nice!

I just have some change requests in how errors are handled. 1) we should use `?` more, and 2) always bubble errors up, so that if something goes wrong that everything fails and we get an error report :)

**Comment by christiangnrd** [comment]
Thanks for the review! I don't have much Rust experience so this is very helpful.

**Review by davidanthoff — APPROVED** [review]

**Pull request merged**